File: extractor-core/src/main/java/ru/mydesignstudio/database/metadata/extractor/extractors/model/TableMetadata.java
package ru.mydesignstudio.database.metadata.extractor.extractors.model;

import java.util.Collection;
import lombok.Data;
import lombok.ToString;

@Data
@ToString
public class TableMetadata {
  private String tableName;
  private String schemaName;
  private Collection<ColumnModel> columns;
  private Collection<PrimaryKeyModel> primaryKeys;
  private Collection<ForeignKeyModel> foreignKeys;
  private Collection<ConstraintModel> constraints;
  private Collection<ReferenceModel> references;
  private Collection<CheckModel> checks;
  private Collection<TriggerModel> triggers;
  private Collection<UdfModel> udfs;
  private Collection<ViewUsedModel> viewsUsed;
  private Collection<ViewReferencedModel> viewsReferenced;
  private Collection<TypeModel> types;
  private Collection<UniqueKeyModel> uniqueKeys;
  private Collection<ScriptModel> scripts;
}

File: extractor-core/src/main/kotlin/ru/mydesignstudio/database/metadata/extractor/extract/result/TableMetadata.kt
package ru.mydesignstudio.database.metadata.extractor.extract.result

data class TableMetadata(
        val tableName: String,
        val schemaName: String,
        val columns: Collection<ColumnModel>,
        val primaryKeys: Collection<PrimaryKeyModel>,
        val foreignKeys: Collection<ForeignKeyModel>,
        val constraints: Collection<ConstraintModel>,
        val references: Collection<ReferenceModel>,
        val checks: Collection<CheckModel>,
        val triggers: Collection<TriggerModel>,
        val udfs: Collection<UdfModel>,
        val viewsUsed: Collection<ViewUsedModel>,
        val viewsReferenced: Collection<ViewReferencedModel>,
        val types: Collection<TypeModel>,
        val uniqueKeys: Collection<UniqueKeyModel>,
        val scripts: Collection<ScriptModel>) {

    companion object {
        @JvmStatic
        fun create(type: TypeModel): TableMetadata {
            return TableMetadata(
                    "",
                    "",
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(type),
                    listOf(),
                    listOf()
            )
        }

        @JvmStatic
        fun create(): TableMetadata {
            return TableMetadata(
                    "",
                    "",
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf(),
                    listOf()
            )
        }
    }
}